pyro/distributions/sparse_multivariate_normal.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+
+import torch
+from torch.autograd import Variable
+from torch.distributions.utils import lazy_property
+
+from pyro.distributions.distribution import Distribution
+from pyro.distributions.util import copy_docs_from, matrix_triangular_solve_compat
+
+
+@copy_docs_from(Distribution)
+class SparseMultivariateNormal(Distribution):
```

Comment:
It would be nice to implement this as a PyTorch distribution + wrapper as we've done with MVN
```py
# in pyro/distributions/torch/sparse_multivariate_normal.py
class TorchSparseMultivariateNormal(torch.distributions.Distribution):
    has_rsample = True
    ...your implementation here...

class SparseMultivariateNormal(TorchDistribution):
    reparameterized = True
    def __init__(self, loc, D_term, W_term, trace_term=None, *args, **kwargs):
        torch_dist = TorchSparseMultivariateNormal(loc, D_term, W_term, trace_term)
        super(SparseMultivariateNormal, self).__init__(torch_dist, *args, **kwargs)
```
This would have a few advantages:
- it makes it easier to move this implementation upstream to PyTorch at some time
- it supports `log_pdf_mask` and `extra_event_dim`
- it makes it easier to register a `kl_divergence()` implementation
- it makes it easier for us to refactor towards PyTorch distributions interface